### PR TITLE
Implement ipvs util package

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

This is a part of work "Implement IPVS-based in-cluster service load balancing"(#46580). 

Since PR #46580 is too big to easily review, I split the original PR and group the commits related to ipvs util package and godeps into this PR in order to make it easier to review.

**Which issue this PR fixes**

#48992

**Special notes for your reviewer**:

This PR is a part of work #46580 and all comment in #46580 has been fixed to this PR. Reviewer can continue to comment in this PR. THANKS!

@thockin @quinton-hoole @bowei @freehan @kevin-wangzefeng @dhilipkumars

**Release note:**:

```release-note
NONE
```

### CODE REVIEW

**Comment by m1093782566** [comment]
ping @thockin

**Comment by m1093782566** [comment]
@kubernetes/sig-network-pr-reviews.

File: pkg/util/ipvs/ipvs.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+package ipvs
+
+import (
+	"net"
+	"strconv"
+)
+
+// Interface is an injectable interface for running ipvs commands.  Implementations must be goroutine-safe.
+type Interface interface {
```
Comment (feiskyer) [inline pkg/util/ipvs/ipvs.go]: change to a more descriptive name?

File: pkg/util/ipvs/ipvs.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	// Flush clears all services in system. return occurred error immediately.
+	Flush() error
+	// EnsureDummyDevice checks if the specified dummy interface is present and, if not, creates it.  If the dummy interface existed, return true.
+	EnsureDummyDevice(dummyDev string) (exist bool, err error)
+	// DeleteDummyDevice deletes the specified dummy interface.  If the dummy interface existed, return error.
```
Comment (feiskyer) [inline pkg/util/ipvs/ipvs.go]: I think if the dummy interface exist, DeleteDummyDevice should delete it instead of returning error.

Comment (m1093782566) [inline pkg/util/ipvs/ipvs.go]: Yes, I agree. Will update the comment message.

File: pkg/util/ipvs/ipvs.go (hunk 1 above)
Comment (m1093782566) [inline pkg/util/ipvs/ipvs.go]: @feiskyer 

I am so glad to receive your feedback. I just find the name in pkg/util/iptables, pkg/util/ipconfig, pkg/util/dbus is `Interface`. I am open to change to a more descriptive name if possible, although I think name `Interface` is more consistent with other util packages.

File: pkg/util/ipvs/ipvs.go (hunk 2 above)
Comment (m1093782566) [inline pkg/util/ipvs/ipvs.go]: Fixed.

**Comment by m1093782566** [comment]
Thanks for your review, @feiskyer. I fixed or replied to your comments inline.

**Comment by kevin-wangzefeng** [comment]
@m1093782566 Can you split Godeps change and imported (/vender) files into a separate commit? That makes the PR easier to review.

**Comment by m1093782566** [comment]
Thanks for your advice. This PR will always keep 2 commits:

* libnetwork ipvs godeps

* ipvs API wrapper util

**Comment by m1093782566** [comment]
I am going to close this PR since we are working on #46580 now

**Pull request closed without merging**